src/query/interpreter.cpp

```diff
@@ -1395,6 +1395,47 @@ auto ParseConfigMap(std::unordered_map<Expression *, Expression *> const &config
          ranges::to<std::map<std::string, std::string, std::less<>>>;
 }
 
+auto ParseVectorIndexConfigMap(std::unordered_map<query::Expression *, query::Expression *> const &config_map,
+                               query::ExpressionVisitor<query::TypedValue> &evaluator)
+    -> storage::VectorIndexConfigMap {
+  if (config_map.empty()) {
+    throw std::invalid_argument("Vector index config map is empty.");
```

Comment:
Add the required field info (capacity and dimension are required + the data types).